editoast/src/views/timetable/stdcm.rs

```diff
@@ -191,6 +186,29 @@ async fn stdcm(
     let earliest_departure_time = stdcm_request.get_earliest_departure_time(simulation_run_time);
     let latest_simulation_end = stdcm_request.get_latest_simulation_end(simulation_run_time);
 
+    let timetable = Timetable::retrieve_or_fail(&mut conn, timetable_id, || {
+        StdcmError::TimetableNotFound { timetable_id }
+    })
+    .await?;
+    // Filter train
+    let mut train_schedules = filter_train_by_start_time_and_timetable(&mut conn, latest_simulation_end, timetable.id).await?;
```

Comment:
Since you already retrieved the timetable, could this be an associated method?

```suggestion
    let mut train_schedules = timetable.schedules_in_time_window(&mut conn, earliest_departure_time, latest_simulation_end).await?;
```

Reply:
Good idea :)